webapp/loggedin/surveyViews/surveyForm/nodeDefs/components/nodeDefErrorBadge.js

```diff
@@ -1,79 +1,91 @@
 import React from 'react'
-import { connect } from 'react-redux'
-
+import { useSelector } from 'react-redux'
+import PropTypes from 'prop-types'
 import * as R from 'ramda'
 
-import ErrorBadge from '@webapp/commonComponents/errorBadge'
-
 import * as Survey from '@core/survey/survey'
 import * as NodeDef from '@core/survey/nodeDef'
 import * as Record from '@core/record/record'
 import * as Node from '@core/record/node'
 import * as RecordValidation from '@core/record/recordValidation'
 import * as Validation from '@core/validation/validation'
 
-import * as SurveyState from '@webapp/survey/surveyState'
-import * as RecordState from '../../../record/recordState'
+import { useSurvey } from '@webapp/commonComponents/hooks'
+import ErrorBadge from '@webapp/commonComponents/errorBadge'
 
-const NodeDefErrorBadge = props => {
-  const { validation, children } = props
+import * as RecordState from '@webapp/loggedin/surveyViews/record/recordState'
 
-  const valid = Validation.isValid(validation)
+const _getValidation = (props) => {
```

Comment:
this could be an external hook useValidation

Reply:
ok